sdk/cosmosdb/cosmos/src/client/ChangeFeed/ChangeFeedForPartitionKey.ts

```diff
@@ -49,6 +50,16 @@ export class ChangeFeedForPartitionKey<T> implements ChangeFeedPullModelIterator
 
   private async instantiateIterator(diagnosticNode: DiagnosticNodeInternal): Promise<void> {
     await this.setIteratorRid(diagnosticNode);
+    if (this.clientContext.enableEncryption) {
+      if (!this.container.isEncryptionInitialized) {
+        await this.container.initializeEncryption();
+      }
+      this.container.encryptionProcessor.containerRid = this.container._rid;
+      this.partitionKey = copyObject(this.partitionKey);
```

Comment:
Add a small comment why this copyObject is required

Reply:
added